torchx/c_src/torchx.cpp

```diff
@@ -21,8 +22,79 @@ inline const std::string* type2string(const torch::ScalarType type)
   return nullptr;
 }
 
+// the class instance to manage the refcount of Tensor
+class TensorP
+{
+public:
+  TensorP(ErlNifEnv *env, const ERL_NIF_TERM arg) : ptr(nullptr)
+  {
+    // setup
+    if (!enif_get_resource(env, arg, TENSOR_TYPE, (void **)&ptr))
+    {
+      // didn't give this parameter
+      return;
+    }
+
+    refcount = (std::atomic<int> *)(ptr + 1);
+
+    if (refcount->load() == 0)
```

Comment:
```suggestion
    if (refcount->load() <= 0)
```

Reply:
Actually I think that the situations in which this could be negative would imply segfaults, right?

Reply:
if this is negative, then there is something terrible wrong.